crates/oxc_traverse/src/context/scoping.rs

```diff
@@ -153,6 +171,28 @@ impl TraverseScoping {
         new_scope_id
     }
 
+    fn insert_scope_below_from_scope_id(
+        &mut self,
+        child_scope_ids: &[ScopeId],
+        scope_id: ScopeId,
+        flags: ScopeFlags,
+    ) -> ScopeId {
+        // Remove these scopes from parent's children
+        if self.scoping.has_scope_child_ids() {
+            self.scoping.remove_child_scopes(scope_id, child_scope_ids);
+        }
+
+        // Create new scope as child of parent
+        let new_scope_id = self.create_child_scope(scope_id, flags);
+
+        // Set scopes as children of new scope instead
+        for &child_id in child_scope_ids {
+            self.scoping.set_scope_parent_id(child_id, Some(new_scope_id));
+        }
+
+        new_scope_id
+    }
```

Comment:
Instead of adding this new method, you could just modify `insert_scope_below` to take a `scope_id` param. That'd reduce repeated code.